examples/c/Makefile

```diff
@@ -106,10 +105,10 @@ $(LIBBLAZESYM_HEADER): $(LIBBLAZESYM_SRC)/target/release/libblazesym.a | $(OUTPU
 	$(Q)cp $(LIBBLAZESYM_SRC)/target/release/blazesym.h $@
 
 # Build BPF code
-$(OUTPUT)/%.bpf.o: %.bpf.c $(LIBBPF_OBJ) $(wildcard %.h) $(VMLINUX) | $(OUTPUT)
+$(OUTPUT)/%.bpf.o: %.bpf.c $(LIBBPF_OBJ) $(wildcard %.h) $(VMLINUX) $(BPFTOOL) | $(OUTPUT)
 	$(call msg,BPF,$@)
-	$(Q)$(CLANG) -g -O2 -target bpf -D__TARGET_ARCH_$(ARCH) $(INCLUDES) $(CLANG_BPF_SYS_INCLUDES) -c $(filter %.c,$^) -o $@
-	$(Q)$(LLVM_STRIP) -g $@ # strip useless DWARF info
+	$(Q)$(CLANG) -g -O2 -target bpf -D__TARGET_ARCH_$(ARCH) $(INCLUDES) $(CLANG_BPF_SYS_INCLUDES) -c $(filter %.c,$^) -o $@.tmp
+	$(Q)$(BPFTOOL) gen object $@ $@.tmp # strip useless DWARF info
```

Comment:
@anakryiko I've very basic knowledge of Makefile.

Your following suggest was not working 
```
$(Q)$(BPFTOOL) gen object $@ $<
```
So I tried following to do something like `bpftool gen object exmaple.bpf.o exmaple.bpf.o` but my assumption is that because of the same name it was corrupting the object file and I was getting `libbpf: failed to get ELF header for .output/minimal.bpf.o: invalid "Elf' handle" error.  
```
$(Q)$(BPFTOOL) gen object $@ $@
``` 
So my workaround is to create temporary object file to generate final object file, e.g. `bpftool gen object exmaple.bpf.o exmaple.bpf.o.tmp` (note: first param is output file and subsequent param(s) are input files), hence `$@.tmp`.

Let me know if there is better way to handle this situation. Thanks.

Reply:
This approach makes sense. I'd like to preserve `.o` suffix, though, so maybe let's use `%.tmp.bpf.o` as intermediate name?